client/asset/bch/spv.go

```diff
@@ -942,20 +942,49 @@ func (w logWriter) Write(p []byte) (n int, err error) {
 // logRotator initializes a rotating file logger.
 func logRotator(netDir string) (*rotator.Rotator, error) {
 	const maxLogRolls = 8
-	logDir := filepath.Join(netDir, logDirName)
-	if err := os.MkdirAll(logDir, 0744); err != nil {
-		return nil, fmt.Errorf("error creating log directory: %w", err)
+	logDir := filepath.Join(filepath.Dir(netDir), logDirName)
+	exist, err := fileExists(logDir)
+	if err != nil {
+		return nil, err
+	}
+	if !exist {
+		oldDir := filepath.Join(netDir, "logs")
+		exist, err := fileExists(oldDir)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
As mentioned in other comment, I'm not certain we need this upgrade check, but if we do, we probably wouldn't bail on an error here. 

We also have `dex.FileExists`, which returns just a bool.

Reply:
Okay. I didn't want to use the dex pkg in the assets pkg to reduce dependency. But maybe it's okay?

Reply:
We're already importing it.